cmake/find/rocksdb.cmake

```diff
@@ -0,0 +1,21 @@
+option(ENABLE_ROCKSDB "Enable ROCKSDB" ${ENABLE_LIBRARIES})
+option(USE_INTERNAL_ROCKSDB_LIBRARY "Set to FALSE to use system ROCKSDB library instead of bundled" ${NOT_UNBUNDLED})
+
+if(ENABLE_ROCKSDB)
+    if (USE_INTERNAL_ROCKSDB_LIBRARY AND NOT MISSING_ROCKSDB)
+        set (ROCKSDB_INCLUDE_DIR "${ClickHouse_SOURCE_DIR}/contrib/rocksdb/include")
+        set (ROCKSDB_CORE_INCLUDE_DIR "${ClickHouse_SOURCE_DIR}/contrib/rocksdb/include")
+        set (ROCKSDB_LIBRARY "rocksdb")
+        set (USE_ROCKSDB 1)
+    else()
+        find_package(ROCKSDB)
+    endif ()
+endif()
+
+if (NOT EXISTS "${ClickHouse_SOURCE_DIR}/contrib/rocksdb/CMakeLists.txt")
+    message (WARNING "submodule contrib is missing. to fix try run: \n git submodule update --init --recursive")
+    set (USE_INTERNAL_ROCKSDB_LIBRARY 0)
+    set (MISSING_ROCKSDB 1)
+endif ()
```

Comment:
AFAIR @abyss7 was against such heuristics, plus it also conflicts with `FAIL_ON_UNSUPPORTED_OPTIONS_COMBINATION` (added in #13687)